# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1652358904 14400
#      Thu May 12 08:35:04 2022 -0400
# Branch stable
# Node ID 8c940cfcce257369677c09154da2aab2c56eaa79
# Parent  63710f3bd9811c2d206ac9e7b4f47cf06c47e153
* scripts/pkg/private/build.m: check configure and Makefile exist before trying to unlink them (Bug #62436)

diff -r 63710f3bd981 -r 8c940cfcce25 scripts/pkg/private/build.m
--- a/scripts/pkg/private/build.m	Wed May 11 09:44:55 2022 -0700
+++ b/scripts/pkg/private/build.m	Thu May 12 08:35:04 2022 -0400
@@ -77,8 +77,12 @@
     else
       arch_abi = getarch ();
       configure_make (desc, build_root, verbose);
-      unlink (fullfile (build_root, "src", "configure"));
-      unlink (fullfile (build_root, "src", "Makefile"));
+      if exist (fullfile (build_root, "src", "configure"), "file")
+        unlink (fullfile (build_root, "src", "configure"));
+      endif
+      if exist (fullfile (build_root, "src", "Makefile"), "file")
+        unlink (fullfile (build_root, "src", "Makefile"));
+      endif
     endif
     tar_name = [desc.name "-" desc.version "-" arch_abi ".tar"];
     tar_path = fullfile (builddir, tar_name);
